### PR TITLE
Upgrade to PostCSS 6 and add the custom parser, stringifier and syntax options

### PR DESCRIPTION
**Pull request by 5310**
I needed to use the SugarSS syntax with PostCSS, and noticed that the plugin didn't support the PostCSS parser options yet. 

I just upgraded the package, and added the option in with a working default. I hope this is acceptable.

### CODE REVIEW

**Comment by 5310** [comment]
I wasn't sure if an expanded example was necessary so I didn't add one beyond the doc.

For reference, I'm using the new option like this:
```js
const Metalsmith = require('metalsmith')
const json = require('jsonfile').readFileSync

new Metalsmith(__dirname)
  .destination('./dist')
  .use(require('metalsmith-babel')(json('.babelrc')))
  .use(require('metalsmith-with-postcss')(
    Object.assign({}, json('.postcssrc'), {
      pattern: ["**/*.sss"],
      parser: require('sugarss')
    })
  ))
  .use(require('metalsmith-rename')([[/\.sss$/, '.css']]))
  .use(require('metalsmith-with-postcss')(
    Object.assign({}, json('.postcssrc'), {
      pattern: ["**/*.css"]
    })
  ))
  .build()
```

**Comment by arccoza** [comment]
Thanks for this pull, the expanded example is a good idea, I will do a complete review and get back to you.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "metalsmith-with-postcss",
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (arccoza) [inline package.json]: Version should be `1.1.0` according to semver rules.

File: index.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ var postcss = require('postcss');
 module.exports = exports = function(options) {
   var defaults = {
     pattern: ['**/*.css', '!**/_*/*', '!**/_*'],
+    parser: undefined,
```
Comment (arccoza) [inline index.js:9]: If you're going to add the parser option, perhaps you should add the `syntax` and `stringifier` options as well.

File: index.js [hunk 3]
```diff
@@ -56,6 +57,7 @@ module.exports = exports = function(options) {
       var promise = processor.process(css, {
         from: path.join(metalsmith._source, key),
         to: path.join(metalsmith._destination, key),
+        parser: options.parser,
```
Comment (arccoza) [inline index.js:62]: `syntax` and `stringifier` options too.

File: package.json [hunk 4]
```diff
@@ -22,6 +22,6 @@
   "homepage": "https://github.com/arccoza/metalsmith-with-postcss",
   "dependencies": {
     "minimatch": "^3.0.0",
-    "postcss": "^5.0.19"
+    "postcss": "^6.0.9"
```
Comment (arccoza) [inline package.json]: A broader version range, `6.x`, so the plugin postcss dependency remains up to date.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
+## [1.0.1] - 2017-08-16
+### Added
+- PostCSS 6.
+- `parser` options.
```
Comment (arccoza) [inline CHANGELOG.md]: `syntax` and `stringifier` options too.

File: README.md [hunk 6]
```diff
@@ -72,6 +72,9 @@ metalsmith(__dirname)
   - `pattern`
     - __Default value:__ `['**/*.css', '!**/_*/*', '!**/_*']`
     - __Description:__ Only process files that match this pattern, can be an array of multiple patterns, following [multimatch](https://github.com/sindresorhus/multimatch) rules. The default patterns exclude any files or folders prefixed with an underscore.
+  - `parser`
```
Comment (arccoza) [inline README.md:79]: Add the expanded code example.

**Review by arccoza — CHANGES_REQUESTED** [review]
This looks good, just a few small tweaks, before merging. To update the pull, just push additional changes to the same branch (`master`) in your fork.

File: package.json (hunk 1 above)
Comment (5310) [reply package.json]: Oops!

File: package.json (hunk 4 above)
Comment (5310) [reply package.json]: What about Minimatch though?

File: index.js (hunk 2 above)
Comment (5310) [reply index.js:9]: Should've. Will do.

File: package.json (hunk 4 above)
Comment (arccoza) [reply package.json]: Minimatch is fine, it's still on 3.0.4, and isn't the important dep. I actually wanted to replace it with [micromatch](https://github.com/micromatch/micromatch) sometime.

File: CHANGELOG.md (hunk 5 above)
Comment (arccoza) [reply CHANGELOG.md]: Don't forget to change the version here as well (1.1.0).

File: package.json [hunk 7]
```diff
@@ -22,6 +22,6 @@
   "homepage": "https://github.com/arccoza/metalsmith-with-postcss",
   "dependencies": {
     "minimatch": "^3.0.0",
-    "postcss": "^5.0.19"
+    "postcss": "^6.x"
```
Comment (arccoza) [inline package.json]: It should be just `6.x` without the `^`.

Comment (arccoza) [reply package.json]: Useful [semver tester](https://semver.npmjs.com/).

Comment (5310) [reply package.json]: That _is_ very useful. Bookmarked!

**Comment by 5310** [comment]
Made the changes. Does the placement of the expanded example work?

**Comment by 5310** [comment]
Newer errors fixed, hopefully.

File: README.md (hunk 6 above)
Comment (arccoza) [reply README.md:79]: You've got parser twice here in the docs, and no `syntax` option.

Comment (5310) [reply README.md:79]: Embarrassing. Fixed.

**Review by 5310 — COMMENTED** [review]
Sorry for all these documentation mistakes. I went over them again and I think they're all covered now.

File: package.json (hunk 7 above)
Comment (arccoza) [reply package.json]: The dep should be simply: `"postcss": "6.x"`

**Review by arccoza — APPROVED** [review]

**Comment by arccoza** [comment]
No worries, and thanks for contributing, I'm going to merge this now.

**Comment by 5310** [comment]
Thanks for the merge! Updating my project to the original NPM module again :D

**Comment by arccoza** [comment]
I've just published the update to NPM, so you should be good.

**Pull request merged**